src/frontend/src/binder/values.rs

```diff
@@ -92,9 +92,17 @@ impl Binder {
         // Calculate column types.
         let types = match expected_types {
             Some(types) => {
+                if types.len() > num_columns {
```

Comment:
Is this check necessary? I guess `cast_on_insert` will do this for us.

Reply:
Yep, you are correct. I am a bit surprised tbh. During development I was running into issues without this check, but I ran it and it seems to be fine.